src/bot/events/message.js

```diff
@@ -6,44 +6,68 @@ const guildMemberModel = require('../models/guild/guildMemberModel.js');
-          return resolve();
-        }
+const acceptedChannelTypes = [
+    'GUILD_TEXT',
+    'GUILD_NEWS',
+    'GUILD_PUBLIC_THREAD'
+]
 
-        if (msg.guild.appData.textXp) { await rankMessage(msg); }
-
-        return resolve();
-      } catch (e) { return reject(e); }
-    });
-  },
+module.exports = {
+	name: 'messageCreate',
+	execute(msg) {
+        return new Promise(async function (resolve,reject) {
+            try {
+                if (msg.author.bot == true || msg.system == true || skip(msg.guildId) || msg.type != 'DEFAULT')
+                    return resolve();
+              
+                if (!msg.guild) {
+                    await msg.reply({ content:('Hi. Please use your commands inside the channel of a server i am in.\n Thanks!'), ephemeral: true });
+                    return resolve();
+                } 
+              
+                await guildModel.cache.load(msg.guild);
+              
+                if (msg.mentions.members.first() && msg.mentions.members.first().id == msg.client.user.id) {
+                    await msg.reply({content:'Hey, thanks for mentioning me! The prefix for the bot on this server is ``'+msg.guild.appData.prefix+'``. Type ``'+msg.guild.appData.prefix+'help`` for more information. Have fun!', ephemeral: true });
+                    return resolve();
+                } 
+                
+              if (msg.content.startsWith(msg.guild.appData.prefix)) {
+                  if (msg.guild.me.permissions.missing('294172224721')) {
+                      if (msg.guild.appData.permsWarningCooldown <= 1) {
+                          msg.guild.appData.permsWarningCooldown = 5
+                          const warning = new MessageEmbed()
+                              .setAuthor('WARNING', 'https://cdn.pixabay.com/photo/2017/03/08/14/20/flat-2126885_1280.png')
+                              .setDescription(`Your bot is missing permissions it needs to function properly! Please ask an administrator or your server owner to [click here to **reinvite it.**](https://discord.com/api/oauth2/authorize?client_id=534589798267224065&permissions=294172224721&scope=bot%20applications.commands)`)
+                              .setColor('#ffcc00')
+                          await msg.channel.send({embeds:[warning]})
+                      } else { 
+                          msg.guild.appData.permsWarningCooldown = (msg.guild.appData.permsWarningCooldown - 1) || 5;
+                      }
+                  }
+                  await handleCommand(msg);
+                  return resolve();
+              }
+              
+              if (msg.guild.appData.textXp && acceptedChannelTypes.includes(msg.channel.type)) { await rankMessage(msg); }
+
+              return resolve();
+
+            } catch (e) { reject(e); }
+        });
+	},
 };
 
 
 function rankMessage(msg) {
   return new Promise(async function(resolve, reject) {
     try {
+      let channel = msg.channel;
```

Comment:
Haven't we patched this already?

Reply:
yeah, but it's forked from master before the patch went into place